### PR TITLE
PLAT-829-1 Removed ActionQueue.ts and Added AjaxRequestMessage.ts

### PR DESCRIPTION
**Pull request by oliver-richers**
- removed Actions.ts and ActionQueue.ts
- introduced new AjaxRequestMessage.ts (no more `any` in TypeScript)
- many other simplifications and clean-up

### CODE REVIEW

File: platform-ajax/src/main/java/com/softicar/platform/ajax/event/AjaxDomEvent.java [hunk 1]
```diff
@@ -53,8 +53,8 @@ public AjaxDomEvent(IDomDocument document, IAjaxRequest ajaxRequest) {
 		this.scrollX = AjaxParameterUtils.getDouble(ajaxRequest, "sx");
 		this.scrollY = AjaxParameterUtils.getDouble(ajaxRequest, "sy");
 		this.keyCode = AjaxParameterUtils.getInteger(ajaxRequest, "k");
-		this.windowWidth = AjaxParameterUtils.getInteger(ajaxRequest, "ww");
-		this.windowHeight = AjaxParameterUtils.getInteger(ajaxRequest, "wh");
+		this.windowWidth = AjaxParameterUtils.getInteger(ajaxRequest, "wx");
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/java/com/softicar/platform/ajax/event/AjaxDomEvent.java:56]: I would have expected a corresponding change on the js/ts side - but there seems to be none. How so?
(same below)

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
 		let data = [];
 
-		for(let key in this.parameters) {
+		for(let [key, value] of this.data) {
 			// push key
 			data.push(key.length+"\n"+key);
 	
 			// push value
-			let value = this.parameters[key];
+			// FIXME this check should be not needed!!!!!!!!!!!!!!!
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts:24]: what about this? when/how should we fix it?

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-ajax/src/main/java/com/softicar/platform/ajax/event/AjaxDomEvent.java (hunk 1 above)
Comment (oliver-richers) [reply platform-ajax/src/main/java/com/softicar/platform/ajax/event/AjaxDomEvent.java:56]: there is, it is, see `AjaxRequestMessage.setPoint`

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts (hunk 2 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts:24]: nice catch, I added this before finishing `AjaxRequestMessage`, now I can say that the check is indeed useless

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts [hunk 3]
```diff
@@ -1,40 +1,31 @@
...
-			}
+			data.push(value.length+"\n"+value);
 		}
 
 		return data.join("");
 	}
 
 	/**
-	 * Encodes the {@link AjaxRequest} parameters into a hexadecimal string.
+	 * Encodes the {@link AjaxRequest} data into a hexadecimal string.
```
Comment (daniel-klose) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts:28]: Shouldnt this be `AjaxRequestMessage`?

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts [hunk 4]
```diff
@@ -1,40 +1,31 @@
...
 	
-	public constructor(parameters: any) {
-		this.parameters = parameters;
+	public constructor(data: Map<string, string>) {
+		this.data = data;
 	}
 
 	/**
-	 * Encodes the {@link AjaxRequest} parameters into a string.
+	 * Encodes the {@link AjaxRequest} data into a string.
```
Comment (daniel-klose) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts:14]: Shouldnt this be `AjaxRequestMessage`?

**Review by daniel-klose — CHANGES_REQUESTED** [review]

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts (hunk 4 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts:14]: yes

**Review by daniel-klose — APPROVED** [review]

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**